optimizely/event_builder.py

```diff
@@ -35,78 +35,81 @@ class BaseEventBuilder(object):
+    """ Get project ID. """
 
-    self.params[self.EventParams.PROJECT_ID] = self.config.get_project_id()
+    return self.config.get_project_id()
 
-  def _add_account_id(self):
-    """ Add account ID to the event. """
+  def _get_account_id(self):
+    """ Get account ID. """
 
-    self.params[self.EventParams.ACCOUNT_ID] = self.config.get_account_id()
+    return self.config.get_account_id()
 
   @abstractmethod
-  def _add_user_id(self, user_id):
-    """ Add user ID to the event.
+  def _get_user_id(self, user_id):
+    """ Get user ID.
 
     Args:
       user_id: ID of the user.
     """
     pass
 
   @abstractmethod
-  def _add_attributes(self, attributes):
-    """ Add attribute(s) information to the event.
+  def _get_attributes(self, attributes):
+    """ Get attribute(s) information.
 
     Args:
       attributes: Dict representing user attributes and values which need to be recorded.
     """
     pass
 
   @abstractmethod
-  def _add_source(self):
-    """ Add source information to the event. """
+  def _get_source(self):
+    """ Get source information. """
     pass
 
-  def _add_revision(self):
-    """ Add datafile revision information to the event. """
+  def _get_revision(self):
+    """ Get datafile revision information. """
     pass
 
-  def _add_anonymize_ip(self):
-    """ Add IP anonymization bool to the event """
+  def _get_anonymize_ip(self):
+    """ Get IP anonymization bool """
 
-    self.params[self.EventParams.ANONYMIZE_IP] = self.config.get_anonymize_ip_value()
+    return self.config.get_anonymize_ip_value()
 
   @abstractmethod
   def _get_time(self):
-    """ Get time in milliseconds to be added to the event.
+    """ Get time in milliseconds to be added.
 
     Returns:
       Current time in milliseconds.
     """
 
     return int(round(time.time() * 1000))
 
-  def _add_common_params(self, user_id, attributes):
-    """ Add params which are used same in both conversion and impression events.
+  def _get_common_params(self, user_id, attributes):
+    """ Get params which are used same in both conversion and impression events.
 
     Args:
       user_id: ID for user.
       attributes: Dict representing user attributes and values which need to be recorded.
     """
+    commonParams = {}
 
-    self._add_project_id()
-    self._add_account_id()
-    self._add_user_id(user_id)
-    self._add_attributes(attributes)
-    self._add_source()
-    self._add_anonymize_ip()
+    commonParams[self.EventParams.PROJECT_ID] = self._get_project_id()
+    commonParams[self.EventParams.ACCOUNT_ID] = self._get_account_id()
+    commonParams[self.EventParams.USERS] = self._get_user_id(user_id)
+    commonParams[self.EventParams.USERS][0][self.EventParams.ATTRIBUTES] = self._get_attributes(attributes)
+    commonParams[self.EventParams.SOURCE_SDK_TYPE] = self._get_source()[self.EventParams.SOURCE_SDK_TYPE]
+    commonParams[self.EventParams.SOURCE_SDK_VERSION] = self._get_source()[self.EventParams.SOURCE_SDK_VERSION]
```

Comment:
it's repetitive. I think you should create methods _getsourcesdktype and _getsourcesdkversion.